### PR TITLE
Refactoring rdkit.Chem.SATIS

### PR DESCRIPTION
**Pull request by gedeck**
Major rewrite of the code. It's probably best to look at the new code and not on the changes.

### CODE REVIEW

File: rdkit/Chem/SATIS.py [hunk 1]
```diff
@@ -12,15 +12,12 @@
```
Comment (greglandrum) [inline rdkit/Chem/SATIS.py]: Doesn't show up here in the diff, but you might as well remove the `$Id$` line here and update the copyright.

**Review by greglandrum — CHANGES_REQUESTED** [review]
A couple small changes requested.
Thanks for doing the complete rewrite and simplifying the code so much.

File: rdkit/Chem/SATIS.py (hunk 1 above)
Comment (gedeck) [inline rdkit/Chem/SATIS.py]: Done

**Comment by gedeck** [comment]
Requested changes implemented

**Review by greglandrum — APPROVED** [review]

**Pull request merged**